### ISSUE TITLE
Add Upload function to Inbox

### ISSUE THREAD

**Issue by Marvo2011** [luke-browning/pdf-web-edit#51]
Hello,

it would be nice to mix documents with a upload button in the inbox.

**Comment by DanielHabenicht** [comment]
I added an upload function in my fork: https://github.com/DanielHabenicht/fork.pdf-web-edit

@luke-browning I am happy to contribute back. I can open a PR if you are interested. Stellar work what you've done so far!

**Comment by luke-browning** [comment]
Hi @DanielHabenicht - this is looking great, a PR would be appreciated!

**Comment by DanielHabenicht** [comment]
Is it ok if I put it all in one PR? There are some additional changes that make developing easier. 

If its not to much at once, otherwise I will have to split it up according to my commits which will take some more time, because it is more work.

**Comment by DanielHabenicht** [comment]
For image upload it would be great if the images would be deskewd and cropped to fit a A4 page. 
I would use https://github.com/dlemstra/Magick.NET for it, is that ok for you? (if you want to have it contributed back)

**Comment by luke-browning** [comment]
In an ideal world it would be easier if you could split the PR's up for each issue/enhancement but if you don't have the time then 1 PR is OK - it just might take me more time to merge!

Is it possible to use the ImageSharp library the project already includes to deskew the image?

**Comment by DanielHabenicht** [comment]
I don't think it is possible with ImageSharp, as I haven't found any documentation indicating it. Maybe I just missed it. If you have a link to that functionality I am happy to have a look.

I will probably send one PR, to much to do and no worries if you take your time.
